Enable usage of `--show-symbol-table` where the symbol table contains unrecognised modes.

thomasspriggs opened the pull request:
Enables usage of `--show-symbol-table` where the symbol table contains unrecognised modes. Relates to https://github.com/diffblue/cbmc/issues/6219
<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.

If your PR fixes a bug, include the regression test(s) in the same commit as the bug fix. Else, keep commits small and orthogonal, possibly placing tests in commits of their own.
--->

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

thomasspriggs committed:
Deduplicate code for getting language for showing symbol table

So that the reader can more easily see that each of the functions in
this file get the language in the same way and so that any future
changes to this functionality only need to be made in one place.

thomasspriggs committed:
Use default language for showing symbols of unrecognised language

thomasspriggs committed:
Document how tests in `symtab2gb` directory work

thomasspriggs committed:
Add tests of --show-symbol-table with unsupported language

To confirm that the fix from the previous "use default language" commit
is worked as expected.

TGWDB reviewed (approved):
Looks good and tested with random and complex language names locally.

peterschrammel commented on src/goto-programs/show_symbol_table.cpp:
```diff
@@ -24,6 +24,17 @@ Author: Daniel Kroening, kroening@kroening.com
 
+/// \brief Gets the language which should be used for showing the type and value
+///   of the supplied \p symbol.
+static std::unique_ptr<languaget>
```
Document the behaviour in the case of unknown or empty modes.

thomasspriggs committed:
Document return value of `get_show_symbol_language` function

Because the behaviour could be surprising if the reader is not familiar
with the intended behaviour.
